main/lsp/LSPTypecheckerCoordinator.cc

```diff
@@ -52,41 +52,43 @@ class TypecheckerTask final : public core::lsp::Task {
 
 class InitializeTask : public LSPTask {
     LSPTypechecker &typechecker;
-    WorkerPool &workers;
     LSPFileUpdates updates;
 
 public:
-    InitializeTask(const LSPConfiguration &config, LSPTypechecker &typechecker, WorkerPool &workers,
-                   LSPFileUpdates updates)
-        : LSPTask(config, true), typechecker(typechecker), workers(workers), updates(move(updates)){};
+    InitializeTask(const LSPConfiguration &config, LSPTypechecker &typechecker, LSPFileUpdates updates)
+        : LSPTask(config, true), typechecker(typechecker), updates(move(updates)){};
 
-    void run(LSPTypecheckerDelegate &_) override {
-        typechecker.initialize(move(updates), workers);
+    void run(LSPTypecheckerDelegate &tcd) override {
+        typechecker.initialize(move(updates), tcd.workers);
     }
 };
 
-class SlowPathTypecheckTask : public LSPTask {
+class SlowPathTypecheckTask : public core::lsp::Task {
```

Comment:
Now that the slow path is non-blocking, this task doesn't need to use the delegate or common functionality in `LSPTask`, so I made it a regular `Task` . 🎉